src/evm_builtins.rs

```diff
@@ -20,109 +20,99 @@ impl Instructions for EVMInstructions {
-                &return_vars.first().unwrap().name,
-                value
-            )
+            smt::define_const(return_vars.first().unwrap().clone(), value)
         };
-        let direct = |smt_name: &str| {
-            let smt_encoding =
-                format!("({} {} {})", smt_name, arguments[0].name, arguments[1].name);
-            single_return(match is_bool_function(smt_name) {
+        let direct = |smt_op: SMTOp| {
+            let smt_encoding = SMTExpr {
+                op: smt_op.clone(),
+                args: arguments.clone().into_iter().map(|v| v.into()).collect(),
+            };
+            single_return(match is_bool_function(smt_op) {
                 true => wrap_boolean(smt_encoding),
                 false => smt_encoding,
             })
         };
 
+        let mut it = arguments.clone().into_iter();
+        let arg_0 = it.next();
+        let arg_1 = it.next();
+
         match builtin.name {
             "stop" => evm_context::set_stopped(ssa),
-            "add" => direct("bvadd"),
-            "sub" => direct("bvsub"),
-            "mul" => direct("bvmul"),
-            "div" => direct("bvudiv"), // TODO check that the parameter oder is correct
-            "sdiv" => direct("bvsdiv"),
-            "mod" => direct("bvurem"),
-            "smod" => direct("bvsmod"), // TODO check if it is bvsmod or bvsrem (they differ in sign)
-            "not" => single_return(format!("(bvnot {})", arguments[0].name)),
-            "lt" => direct("bvult"),
-            "gt" => direct("bvugt"),
-            "slt" => direct("bvslt"),
-            "sgt" => direct("bvsgt"),
-            "eq" => direct("="),
-            "iszero" => single_return(wrap_boolean(format!(
-                "(= {} #x0000000000000000000000000000000000000000000000000000000000000000)",
-                arguments[0].name
-            ))),
-            "and" => direct("bvand"),
-            "or" => direct("bvor"),
-            "xor" => direct("bvxor"),
+            "add" => direct(SMTOp::BvAdd),
+            "sub" => direct(SMTOp::BvSub),
+            "mul" => direct(SMTOp::BvMul),
+            "div" => direct(SMTOp::BvUDiv), // TODO check that the parameter oder is correct
+            "sdiv" => direct(SMTOp::BvSDiv),
+            "mod" => direct(SMTOp::BvURem),
+            "smod" => direct(SMTOp::BvSMod), // TODO check if it is bvsmod or bvsrem (they differ in sign)
+            "not" => single_return(smt::bvnot(arg_0.unwrap().into())),
+            "lt" => direct(SMTOp::BvULt),
+            "gt" => direct(SMTOp::BvUGt),
+            "slt" => direct(SMTOp::BvSLt),
+            "sgt" => direct(SMTOp::BvSGt),
+            "eq" => direct(SMTOp::Eq),
+            "iszero" => single_return(wrap_boolean(smt::eq_zero(arg_0.unwrap().into()))),
+            "and" => direct(SMTOp::BvAnd),
+            "or" => direct(SMTOp::BvOr),
+            "xor" => direct(SMTOp::BvXor),
             "byte" => {
-                let byte_index = &arguments[0].name;
```

Comment:
Maybe more readable to keep these locals?

Reply:
It's still there as
```rust
let byte_index: SMTExpr = arg_0.unwrap().into();`
```